fiftyone/operators/executor.py

```diff
@@ -146,12 +147,18 @@ def execute_operator(operator_uri, ctx=None, **kwargs):
     """
     request_params = _parse_ctx(ctx=ctx, **kwargs)
 
-    result = asyncio.run(
-        execute_or_delegate_operator(
-            operator_uri, request_params, exhaust=True
-        )
+    coroutine = execute_or_delegate_operator(
+        operator_uri, request_params, exhaust=True
     )
-    result.raise_exceptions()
+
+    if foc.is_notebook_context():
```

Comment:
are there other cases that an event loop is existing? instead of being notebook specific we could do:

```python
try:
    loop = asyncio.get_running_loop()
    result = loop.run_until_complete(coro)
except RuntimeError:
    result = asyncio.run(coro)
```

Reply:
since this is a less common case, maybe we don't want to rely on try-except

Reply:
```python
try:
    loop = asyncio.get_running_loop()
    result = loop.run_until_complete(coro)
except RuntimeError:
    result = asyncio.run(coro)
``` 
I was going to recommend the above as well. `get_event_loop` is deprecated I believe. The above handles any situation where an event loop is running, and falls back to `run` if not

Reply:
I love the idea of `try-except` around `asyncio.get_running_loop()` rather than specifically checking for notebook contexts 👍 

Sadly, `run_until_complete()` is not allowed in Jupyter notebooks 😭 
https://nocomplexity.com/documents/jupyterlab/tip-asyncio.html

Reply:
Hmm, perhaps the best practice is the user should await the coroutine in their notebook. From [stackoverflow](https://stackoverflow.com/questions/47518874/how-do-i-run-python-asyncio-code-in-a-jupyter-notebook/53581739#53581739)

Reply:
Yeah I arrived at the same conclusion and updated the PR. It's fine to just return the task to the caller and then they can `await` it in the notebook.

Will update the docs accordingly too 👍

Reply:
> Sadly, run_until_complete() is not allowed in Jupyter notebooks 😭
😭 😠